src/main/scala/nl/vroste/zio/kinesis/client/Producer.scala

```diff
@@ -128,10 +128,14 @@ object Producer {
       failedQueue     <- zio.Queue.bounded[ProduceRequest](settings.bufferSize).toManaged(_.shutdown)
 
       triggerUpdateShards <- Util.periodicAndTriggerableOperation(
-                               (log.debug("Refreshing shard map") *>
-                                 (getShardMap(streamName) >>= currentShardMap.set) *>
-                                 log.info("Shard map was refreshed")).orDie,
-                               settings.updateShardInterval
+                               {
+                                 log.debug("Refreshing shard map") *>
+                                   (getShardMap(streamName) >>= currentShardMap.set) *>
+                                   log.info("Shard map was refreshed")
+                               }.tapError { e =>
```

Comment:
Finally catching up and updating to this version... I think this should be `catchAll` instead, otherwise the error is re-raised and still end up in the output, just with an additional line for the proper logging. @svroonland this seems to be the case on master too, what do you say about updating this in both branches? (using `catchError` or moving the `ignore` one level up)

Reply:
Yeah, the ignore should be one level up. You shouldn't see the bubbled up error until the Producer (fiber) is shutdown though, right?

Reply:
Like so? https://github.com/svroonland/zio-kinesis/pull/543

Reply:
#543 looks alright.

I have a producer that stays alive as long as the app does, and I can see more logs after this error is dumped into stderr, so I think the producer is supposedly still running.

Seeing that `periodicAndTriggerableOperation` does `((queue.take raceFirst ZIO.sleep(period)) *> effect *> queue.takeAll).forever.forkManaged`, wouldn't this mean that the first failure/death in `effect` would kill the whole fiber and stop refreshing the shard map forever?

Reply:
You are right. It would be better to make `periodicAndTriggerableOperation` take a `ZIO[R, Nothing, A]` (URIO)

Reply:
That could work, yeah. But then there's the rick of an uncaught failure in the untyped channel 😅

Reply:
You mean a defect / ZIO.die ?

Reply:
Yeah